### PR TITLE
Bamboo collector

### PR DESCRIPTION
**Pull request by raynaya**
Bamboo collector, very similar to the Jenkins collector with minor changes

* Captures jobs under a plan and under the sub-plans based on different branches
* Cannot capture build log as it is not provided in the bamboo api

### CODE REVIEW

File: Bamboo-build-collector/src/test/java/com/capitalone/dashboard/collector/DefaultBambooClientTests.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        settings = new BambooSettings();
+        bambooClient = defaultBambooClient = new DefaultBambooClient(restOperationsSupplier,
+                settings);
+    }
+
+    @Test
+    public void joinURLsTest() throws Exception {
+        String u = DefaultBambooClient.joinURL("http://jenkins.com",
+                "/api/json?tree=jobs[name,url,builds[number,url]]");
+        assertEquals("http://jenkins.com/api/json?tree=jobs[name,url,builds[number,url]]", u);
```
Comment (tabladrum) [inline Bamboo-build-collector/src/test/java/com/capitalone/dashboard/collector/DefaultBambooClientTests.java]: not that it matters... change the test urls to bamboo?

File: Bamboo-build-collector/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    #Database Password - default is blank
+    dbpassword=dbpass
+
+    #Collector schedule (required)
+    bamboo.cron=0 0/5 * * * *
+
+    #Jenkins server (required) - Can provide multiple
+    bamboo.servers[0]=http://jenkins.company.com
+
```
Comment (tabladrum) [inline Bamboo-build-collector/README.md]: change it to bemboo.company.com

**Comment by tabladrum** [comment]
Also, can you attach some screen shots of your dashboard with Bamboo jobs? and details? Thanks for the PR. Great job.

**Comment by raynaya** [comment]
screen shot 2016-08-30 at 1 13 05 am

screen shot 2016-08-30 at 1 11 29 am

Updated the pr, please check and let me know.  Also check if the screenshots are good enough or not. Thanks!

**Comment by raynaya** [comment]
This collector works just like Jenkins. The build process is a little different here. They have plans concept (equivalent to jobs) and you can have sub-plans(based on branches). This collector will find the high level plans and give you the list of high level plans available. If you select any high-level plan it will fetch all the jobs under that plan (recursively going inside sub-plans if any present based on branches ) .

**Comment by tabladrum** [comment]
LGTM

**Comment by amitmawkin** [comment]
LGTM

**Pull request merged**